docgen/docgen.go

```diff
@@ -52,18 +53,33 @@ func (p *docPlugin) Generate(req *plugin_go.CodeGeneratorRequest) (*plugin_go.Co
 		}
 	}
 
-	var source *google_protobuf.FileDescriptorProto
+	var source *parser.FileDescWrapper
 	for _, f := range req.GetProtoFile() {
 		if strings.Contains(f.GetName(), "all.proto") {
-			source = f
-			break
+			for _, fileToGenerate := range req.FileToGenerate {
+				if fileToGenerate == f.GetName() {
```

Comment:
why is the extra loop necessary? you don't even use fileToGenerate for anything, as far as I can see

Reply:
This checks if the file in the outer loop `f` is part of the descriptors filesToGenerate. Before this, it simply chose the first in the ProtoFile, which is usually a dependency in this case @mvdan

Reply:
Ok, gotcha - please document this.

Reply:
i would suggest 

```go
var source *parser.FileDescWrapper
for _, ftg := range req.FileToGenerate {
	if !strings.Contains(f.GetName(), "all.proto") {
             continue
        }
        source = f
        break
}
```
no need to iterate over ProtoFile at this point, since you're just looking for the file we actually want to generate

https://godoc.org/github.com/golang/protobuf/protoc-gen-go/plugin#CodeGeneratorRequest

Reply:
Oh ok thanks for this @mgfcmartinez

Reply:
@mgfcmartinez the ftg, in this case, will be a string but the source has to be/have a FileDescriptorProto though.

Reply:
@kofoworola right- my bad. you can iterate over req.ProtoFile instead, since it should also contain the file to generate. you could also take the last element in the slice, since we [sort](https://github.com/gunk/gunk/blob/4587cc79bbf6903ef45ef837099dbf3714cde8ad/generate/generate.go#L320) it beforehand

Reply:
I think taking the last element is better, skipped my mind topologicalSort already sorts it

Reply:
but then again there needs to be check for if the FilesToGenerate is empty, so i would leave it

Reply:
Also @mgfcmartinez after running it a few times, there were some scenarios where the FileToGenerate was not the last protofile. Perhaps i could create an issue for that later on